Orbeon patch for Odoo version 12.

Odoo no longer stores the password in the session, this is a problem for the Single Sign On function we have created for the Orbeon module.

We therefor apply a patch on the 'http.py' of Odoo. The patch is different from the patch provided with the version 10 module.

We make these changes by hand, and whenever we apply an update (git pull) we use the git stash / git stash apply function.

We use 'git stash' before we pull Odoo, otherwise git will block the pull request due to uncommited changes.

After the pull is completed we use 'git stash apply' to re-apply the patch.

This is the patch:

diff --git a/odoo/http.py b/odoo/http.py
index d5650e4e..4e755e02 100644
--- a/odoo/http.py
+++ b/odoo/http.py
@@ -1043,6 +1043,7 @@ class OpenERPSession(werkzeug.contrib.sessions.Session):
         self.db = db
         self.uid = uid
         self.login = login
+        self.password = password
         self.session_token = uid and security.compute_session_token(self, request.env)
         request.uid = uid
         request.disable_db = False
@@ -1076,6 +1077,7 @@ class OpenERPSession(werkzeug.contrib.sessions.Session):
         self.setdefault("db", None)
         self.setdefault("uid", None)
         self.setdefault("login", None)
+        self.setdefault("password", None)
         self.setdefault("session_token", None)
         self.setdefault("context", {})
